File: LeetCode_Marathons/LCM1/Medium/K_Closest_Points_to_Origin/a.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        int n = points.size();
        vector<vector<int>> listDistance;

        for (int i = 0; i < n; i++) {
            int distance = points[i][0] * points[i][0] + points[i][1] * points[i][1];
            listDistance.push_back({distance, i});
        }

        sort(listDistance.begin(), listDistance.end());

        vector<vector<int>> ans;
        for (int i = 0; i < k; i++) ans.push_back(points[listDistance[i][1]]);

        return ans;
    }
};

File: LeetCode_Marathons/LCM1/Medium/K_Closest_Points_to_Origin/a.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        int n = points.length;
        Integer[][] listDistance = new Integer[n][2];
        
        for (int i = 0; i < n; i++) {
            int distance = points[i][0] * points[i][0] + points[i][1] * points[i][1];
            listDistance[i] = new Integer[]{distance, i};
        }
        
        Arrays.sort(listDistance, new Comparator<Integer[]>() {
            public int compare(Integer[] a, Integer[] b) {
                return a[0].compareTo(b[0]);
            }
        });

        int[][] ans = new int[k][2];
        for (int i = 0; i < k; i++) ans[i] = points[listDistance[i][1]];

        return ans;
    }
}